wazo_router_confd/auth.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional, List
+
+from aiohttp import ClientSession
+from aiohttp import TCPConnector  # type: ignore
+
+from fastapi import FastAPI
+from starlette.requests import Request
+from starlette.responses import Response
+
+
+X_AUTH_TOKEN_HEADER = 'X-Auth-Token'
+WAZO_TENANT = 'Wazo-Tenant'
+
+
+class Principal(object):
+    auth_id: str
+    uuid: str
+    tenant_uuid: str
+    tenant_uuids: List[str]
+    token: str
+
+    def __init__(
+        self,
+        auth_id: str,
+        uuid: str,
+        tenant_uuid: str,
+        tenant_uuids: List[str],
+        token: str,
+    ):
+        self.auth_id = auth_id
+        self.uuid = uuid
+        self.tenant_uuid = tenant_uuid
+        self.tenant_uuids = tenant_uuids
+        self.token = token
+
+    def __str__(self):
+        return (
+            '<Principal auth_id=%s uuid=%s tenant_uuid=%s tenant_uuids=%s token=%s>'
+            % (self.auth_id, self.uuid, self.tenant_uuid, self.tenant_uuids, self.token)
+        )
+
+
+class AuthClient(object):
+    def __init__(self, url: str, cert: bool):
+        self._url = url
+        self._cert = cert
+
+    async def get_token_data(self, token: str, tenant_uuid: str) -> Optional[Principal]:
+        verify_ssl = bool(self._cert)
+        connector = TCPConnector(verify_ssl=verify_ssl)
+        async with ClientSession(connector=connector) as session:
+            token_data = dict()
+            token_url = self._url + '/token/' + token
+            async with session.get(token_url) as response:
+                if response.status == 200:
+                    data = await response.json()
+                    if data is not None and data.get('data'):
+                        token_data.update(
+                            dict(
+                                auth_id=data['data']['auth_id'],
+                                uuid=data['data']['metadata']['uuid'],
+                                tenant_uuid=data['data']['metadata']['tenant_uuid'],
+                                tenant_uuids=[data['data']['metadata']['tenant_uuid']],
+                                token=data['data']['token'],
+                            )
+                        )
+            if not token_data:
+                return None
```

Comment:
I think a `401` should be returned here if the GET does not return a `200`

Reply:
Right, done!